scipy/stats/tests/test_morestats.py

```diff
@@ -301,6 +305,71 @@ def test_gumbel_r(self):
         assert_array_less(A1, crit1[-2:])
         assert_(A2 > crit2[-1])
 
+    def test_weibull_min_case_A(self):
+        # data and reference values from `anderson` reference [7]
+        x = np.array([225, 171, 198, 189, 189, 135, 162, 135, 117, 162])
+        res = stats.anderson(x, 'weibull_min')
+        m, loc, scale = res.fit_result.params
+        assert_allclose((m, loc, scale), (2.38, 99.02, 78.23), rtol=2e-3)
+        assert_allclose(res.statistic, 0.260, rtol=1e-3)
+        assert res.statistic < res.critical_values[0]
+
+        c = 1 / m  # ~0.42
+        assert_allclose(c, 1/2.38, rtol=2e-3)
+        # interpolate between rows for c=0.4 and c=0.45, indices -3 and -2
+        As40 = _Avals_weibull[-3]
+        As45 = _Avals_weibull[-2]
+        As_ref = As40 + (c - 0.4)/(0.45 - 0.4) * (As45 - As40)
+        # atol=1e-3 because results are rounded up to the next third decimal
+        assert np.all(res.critical_values > As_ref)
+        assert_allclose(res.critical_values, As_ref, atol=1e-3)
+
+    def test_weibull_min_case_B(self):
+        # From `anderson` reference [7]
+        x = np.array([74, 57, 48, 29, 502, 12, 70, 21,
+                      29, 386, 59, 27, 153, 26, 326])
+        res = stats.anderson(x, 'weibull_min')
+        assert_allclose(res.fit_result.params, (0.763, 9.313, 93.5), rtol=1e-3)
+        assert_allclose(res.statistic, 0.54, rtol=1e-1)
+        assert res.statistic > res.critical_values[0]
+
+    def test_weibull_error(self):
+        # Check for helpful error message when Anderson test for Weibull fails
+        rng = np.random.default_rng(2044208699473992917)
+        c, loc, scale = 5, 0, 1
+        x = stats.weibull_min.rvs(c, loc, scale, size=100, random_state=rng)
+        message = "An error occured while fitting the Weibull distribution..."
+        with (pytest.raises(ValueError, match=message),
+              pytest.raises(FloatingPointError)):
```

Comment:
I think this is the cause of the test failures. I never remember how to match multiple error types correctly in Python 3.8 (and before).